src/rrt/Tree.hpp

```diff
@@ -309,7 +372,14 @@ class Tree {
         }
 
         // Add a node to the tree for this state
-        _nodes.push_back(Node<T>(intermediateState, source));
+        Node<T> newNode =
+            Node<T>(intermediateState, source, _dimensions, _TToArray);
+        flann::Matrix<double>* point =
+            new flann::Matrix<double>(newNode.coordinates(), 1, _dimensions);
+        _kdtree.addPoints(*point);
```

Comment:
Can point just be a normal object here since you aren't dealing with the address?